DQMOffline/Trigger/plugins/PhotonMonitor.cc

```diff
@@ -0,0 +1,351 @@
+  photonME_variableBinning_.denominator = nullptr;
+  photonVsLS_.numerator   = nullptr;
+  photonVsLS_.denominator = nullptr;
+  photonEtaME_.numerator   = nullptr;
+  photonEtaME_.denominator = nullptr;
+  photonPhiME_.numerator   = nullptr;
+  photonPhiME_.denominator = nullptr;
+  photonEtaPhiME_.numerator   = nullptr;       
+  photonEtaPhiME_.denominator = nullptr;
+  photonr9ME_.numerator   = nullptr;       
+  photonr9ME_.denominator = nullptr;
+  photonHoverEME_.numerator   = nullptr;       
+  photonHoverEME_.denominator = nullptr;
+  
+}
+
+PhotonMonitor::~PhotonMonitor()
+{
+  if (num_genTriggerEventFlag_) delete num_genTriggerEventFlag_;
+  if (den_genTriggerEventFlag_) delete den_genTriggerEventFlag_;
+}
+
+MEbinning PhotonMonitor::getHistoPSet(edm::ParameterSet pset)
+{
+  return MEbinning{
+    pset.getParameter<int32_t>("nbins"),
+      pset.getParameter<double>("xmin"),
+      pset.getParameter<double>("xmax"),
+      };
+}
+
+MEbinning PhotonMonitor::getHistoLSPSet(edm::ParameterSet pset)
+{
+  return MEbinning{
+    pset.getParameter<int32_t>("nbins"),
+      0.,
+      double(pset.getParameter<int32_t>("nbins"))
+      };
+}
+
+void PhotonMonitor::setTitle(PhotonME& me, std::string titleX, std::string titleY)
+{
+  me.numerator->setAxisTitle(titleX,1);
+  me.numerator->setAxisTitle(titleY,2);
+  me.denominator->setAxisTitle(titleX,1);
+  me.denominator->setAxisTitle(titleY,2);
+
+}
+
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbins, double min, double max)
+{
+  me.numerator   = ibooker.book1D(histname+"_numerator",   histtitle+" (numerator)",   nbins, min, max);
+  me.denominator = ibooker.book1D(histname+"_denominator", histtitle+" (denominator)", nbins, min, max);
+}
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, const std::vector<double>& binning)
+{
+  int nbins = binning.size()-1;
+  std::vector<float> fbinning(binning.begin(),binning.end());
+  float* arr = &fbinning[0];
+  me.numerator   = ibooker.book1D(histname+"_numerator",   histtitle+" (numerator)",   nbins, arr);
+  me.denominator = ibooker.book1D(histname+"_denominator", histtitle+" (denominator)", nbins, arr);
+}
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, double ymin, double ymax)
+{
+  me.numerator   = ibooker.bookProfile(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, ymin, ymax);
+  me.denominator = ibooker.bookProfile(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, ymin, ymax);
+}
+void PhotonMonitor::bookME(DQMStore::IBooker &ibooker, PhotonME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax)
```

Comment:
These method overloads are very hard to follow, why not just name them for what they are supposed to do?
bookME2D or book2DME or bookProfileME

The main worry of mine is the added memory consumption.